src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -134,7 +134,14 @@
      * @param exerciseId the exercise id we are interested in
      * @return the number of distinct submissions belonging to the exercise id
      */
-    @Query("SELECT COUNT (DISTINCT p) FROM ProgrammingExerciseStudentParticipation p WHERE p.exercise.id = :#{#exerciseId} AND EXISTS (SELECT s FROM ProgrammingSubmission s WHERE s.participation.id = p.id AND s.submitted = TRUE) AND NOT EXISTS (select prs from p.results prs where prs.assessor.id = p.student.id)")
+    @Query("""
+            SELECT COUNT (DISTINCT p) FROM ProgrammingExerciseStudentParticipation p
+            WHERE p.exercise.id = :#{#exerciseId}
+            AND (p.testRun = FALSE OR p.testRun IS NULL)
```

Comment:
```suggestion
            AND p.testRun = FALSE
```

Reply:
the value cannot be null, and OR can be significantly slower in such cases

Reply:
fixed in 139c8fb547deadc2b10c0e497d91389dcb5f736c